### DISCUSSION TITLE
[Bug]: The security token included in the request is invalid

### DISCUSSION THREAD

**Discussion by BrachyKnight** [bbc/sqs-consumer#368] [Help] [answered]
### Describe the bug

Hi,
sorry to bother you but I've been indirectly using your package by using [nestjs-sqs](https://github.com/ssut/nestjs-sqs).
Since that package was using a version of sqs-consumer working with the AWS SDK v2, I have decided to write my implementation by using only your most recent package.
Before I was using [alpine-sqs](https://github.com/roribio/alpine-sqs) image as a mock sqs server with invalid credentials. However since switching to v3 of the AWS SDK the consumer returns `SQSError: SQS receive message failed: The security token included in the request is invalid`. I have also tried to use valid credentials, but this time it seems like it doesn't look for the queue_url that i pass in `Consumer.create()` and returns `SQSError: SQS receive message failed: The address https://sqs.eu-south-1.amazonaws.com/ is not valid for this endpoint`.
Is your package running a credential check under the hood? Or is it the sdk v3? I have not found informations regarding that anywhere.

### Your minimal, reproducible example

I have provided the necessary files

### Steps to reproduce

1. Try passing invalid credentials to `Consumer.create()`
2. Try passing valid credentials but a totally different queuUrl in `Consumer.create()` (like 'http://localhost:9324/queue/queue-name')
3. Create a basic node app that uses the environment variables included in the .env I provided. I have also added a docker-compose that relies on that dockerfile to create a network with the mock SQS sever
[issue.zip](https://github.com/bbc/sqs-consumer/files/10823775/issue.zip)

### Expected behavior

I expected the credentials to work even if not valid as in the SDK v2.
If that didn't work I expected the app to use the queue_url I provided

### How often does this bug happen?

None

### Platform

- Dockerized Environment using latest node image and [alpine-sqs](https://github.com/roribio/alpine-sqs)

### Package version

7.0.1

### AWS SDK version

3.276

**Comment by nicholasgriffintn** [comment, marked as answer]
Hey, I've converted this to a discussion as this is not an issue with the sqs-consumer library, we don't do anything with credentials, that is managed by the SQSClient, either the one you provide, or the default client within the codebase.

So this is an issue with the queue that you have setup, your credentials or how you are providing those credentials.

By default, sqs-consumer expects credentials to be provided like this:

https://github.com/bbc/sqs-consumer#credentials

Which also details how you can provide a custom SQSClient. For a custom endpoint, where you are not using an SQS queue, you will have to supply a custom endpoint, like we do here: https://github.com/bbc/sqs-consumer/blob/main/test/features/utils/consumer/handleMessage.js, the custom sqs client for that is setup here: https://github.com/bbc/sqs-consumer/blob/main/test/features/utils/sqs.js.

**Reply by BrachyKnight** [reply]
Thank you, I didn't know for sure if the error was from the consumer or the AWS SDK. I didn't know you had to pass the endpoint config param inside the SQSClient constructor, I assumed that the queueUrl was enough.
Thank you very much for the fast reply!

**Reply by nicholasgriffintn** [reply]
No problem! Feel free to update this discussion if you need anything else to do with this.

Happy to help you get integrated :).
